### ISSUE TITLE
warmup example in the document conflicts with lr scheduler and lr monitor

### ISSUE THREAD

**Issue by zehongs** [PyTorchLightning/pytorch-lightning#9600]
### To Reproduce

The warm-up example in the [document](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#optimizer-step) suggests to manually warm-up lr before optimizer step. 
```
# learning rate warm-up
def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
    # warm up lr
    if self.trainer.global_step < 500:
        lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
        for pg in optimizer.param_groups:
            pg['lr'] = lr_scale * self.hparams.learning_rate

    # update params
    optimizer.step()
    optimizer.zero_grad()
```

However, in the `training_epoch_loop.py`, the scheduler updater will change the lr. 
https://github.com/PyTorchLightning/pytorch-lightning/blob/381343a79c703f2ccf1ab7c1d87400ad6e31fdf4/pytorch_lightning/loops/epoch/training_epoch_loop.py#L137-L150

I think this will not affect the warm-up behavior of optimizer, since the updated lr is never used.
But, will it affect the lr monitor? According to the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.callbacks.lr_monitor.html?highlight=lr), the monitor is call on the start of the batch/epoch. 

**So during warm-up stage, a wrong lr is recorded at the beginning of the next batch-loop.**

### Expected behavior

1. Correct warm-up learning-rate to be recorded.
2. Suggestion for better warmp-up code style. Update documentation.

### Environment

- PyTorch Lightning Version (1.4.5)

### Additional context

Related issue https://github.com/PyTorchLightning/pytorch-lightning/issues/4902

**Comment by rohitgr7** [comment]
the example seems incorrect. warmup should happen after the optimizer step so that it's effect will begin from the 2nd batch. Mind send a PR to move flip it?

**Comment by tchaton** [comment]
Dear @zehongs,

Yes this was a silly example. It is much better to use a lr_scheduler for this use-case.

Best,
T.C
